csrc/device_lower/pass/allocation.cpp

```diff
@@ -479,11 +479,21 @@ class AllocationInserter : public kir::ExprMutator {
     // Fill in the base address, lane offset, and column offset for tensor
     // memory allocations
     if (memory_type == MemoryType::Tensor) {
-      auto allocation_address =
-          GpuLower::current()->tmemInfo().allocation_address;
-      auto address_ti = IrBuilder::create<kir::TensorIndex>(
-          allocation_address, allocation_address->fusion()->zeroVal());
-      alloc_expr->setAddress(address_ti);
+      const auto& regions = GpuLower::current()->tmemInfo().allocation.regions;
+      for (const auto& region : regions) {
```

Comment:
Should we assert a covering region is indeed found? IIUC, it should not happen as `computeTmemInfo` looks at all the tensors of the fusion, but it seems like a good error check to do here.

Reply:
Added